server/api/level.go

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package api
+
+import (
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+)
+
+type Level int
+
+var _ pflag.Value = (*Level)(nil)
+
+const (
+	Beginner = Level(1) + iota
+	Intermediate
+	Advanced
+	Expert
+)
+
+const (
+	LevelAdvanced           = "advanced"
+	LevelAdvancedSymbol     = "◈"
+	LevelBeginner           = "beginner"
+	LevelBeginnerSymbol     = "◉"
+	LevelExpert             = "expert"
+	LevelExpertSymbol       = "◈◈"
+	LevelIntermediate       = "intermediate"
+	LevelIntermediateSymbol = "▣"
+)
+
+func (level Level) String() string {
```

Comment:
I think not. The `Stringer` interface should be defined on a value so that it can be invoked on one. Otherwise `(&l).String()` would be required and it wouldn't work on just a `Level(1)`.